Clarification about client use public ip property (#1108)
hazelcast/hz-docs: docs/modules/clients/pages/java.adoc

The client executes each operation through the already established connection to the cluster.
If this connection(s) disconnects or drops, the client tries to reconnect as configured.

> The initial connection of a client will be established using one of the addresses you provide in the <<configuring-address-list, address list>>. 

Reviewer: this is not correct! unisocket also uses the configured list.

> The initial connection of a client will be established using one of the addresses you provide in the <<configuring-address-list, address list>>. 

Reviewer: sorry, mixed with other comment.

> In case your client is <<java-client-operation-modes, smart>>, the client will then get the addresses of other members in the cluster and try to connect to them.

Reviewer: not correct! unisocket gets the members as well but only connects to one member.

> In case your client is <<java-client-operation-modes, smart>>, the client will then get the addresses of other members in the cluster and try to connect to them.

Author: ok this is acceptable, will reword it

> In case your client is <<java-client-operation-modes, smart>>, the client will then get the addresses of other members in the cluster and try to connect to them.

Author: https://github.com/hazelcast/hz-docs/pull/1108/commits/4ea9d9f7d6b4ba94436650831011aa4c27bfaa50

> If your client is <<java-client-operation-modes, smart>>, the client tries to connect to them.

Author: this is not important though, I am mentioning a scenario that only applies to smart

> If your client is <<java-client-operation-modes, smart>>, the client tries to connect to them.

Author: @ihsandemir please note that "them" in this sentence means other members other than initial member

> If your client is <<java-client-operation-modes, smart>>, the client tries to connect to them.

Author: which means we cant say "if the client is in unisocket mode, it will connect to only one of them in a random way." because it's wrong

> If any of those members has private & public addresses at the same time, the client needs to decide

Author: https://github.com/hazelcast/hz-docs/pull/1108/commits/4ea9d9f7d6b4ba94436650831011aa4c27bfaa50

> which one it should use to connect to those members. You may want to override the property `hazelcast.discovery.public.ip.enabled`

Author: https://github.com/hazelcast/hz-docs/pull/1108/commits/4ea9d9f7d6b4ba94436650831011aa4c27bfaa50

> which one to use when connecting to those members. You can override the property `hazelcast.discovery.public.ip.enabled` property 

Reviewer 2: ```suggestion
which one to use when connecting to those members. You can override the `hazelcast.discovery.public.ip.enabled` property 
```

> depending on your setup. See the <<client-system-properties, Client System Properties section>> for the description of 

Reviewer 2: ```suggestion
to suit your setup. For further information on the `hazelcast.discovery.public.ip.enabled` property, see <<client-system-properties, Client System Properties>>. 
```

Author: https://github.com/hazelcast/hz-docs/pull/1108/commits/4ea9d9f7d6b4ba94436650831011aa4c27bfaa50

> the `hazelcast.discovery.public.ip.enabled` property. 

Reviewer 2: ```suggestion

```

Author: https://github.com/hazelcast/hz-docs/pull/1108/commits/4ea9d9f7d6b4ba94436650831011aa4c27bfaa50

> see <<client-system-properties, Client System Properties section>>.

Reviewer 2: ```suggestion
see <<client-system-properties, Client System Properties>>.
```

> In case you are not providing an address list,

Reviewer 2: ```suggestion
If you use a <<client-network,discovery mechanism>> to find the initial member for the connection instead of an address list,
```

Author: https://github.com/hazelcast/hz-docs/pull/1108/commits/03baba7f36998a8bb8979e6f6122aac48fe7aa20

> but using a <<client-network, discovery mechanism>> to find the initial member to connect to,

Reviewer 2: ```suggestion

```

Author: https://github.com/hazelcast/hz-docs/pull/1108/commits/4ea9d9f7d6b4ba94436650831011aa4c27bfaa50

> above property also applies to the initial member connection private vs public address selection decision.

Reviewer 2: ```suggestion
you can use the same property to configure whether the initial member connection uses the private or public address.
```

> For an example of this scenario, you can have a look at 

Reviewer 2: ```suggestion
For an example of this scenario, refer to
```

Author: 14bc3591

> xref:hazelcast-platform-operator-expose-externally.adoc[Connect to Hazelcast from Outside Kubernetes] guide.

Reviewer 2: ```suggestion
xref:hazelcast-platform-operator-expose-externally.adoc[Connect to Hazelcast from Outside Kubernetes, window=_blank] in the Operator documentation.
```

See xref:extending-hazelcast:discovery-spi.adoc[Discovery SPI] for more information.

|`hazelcast.discovery.public.ip.enabled`
|null
|bool

> |Overrides client behaviour when choosing between using public or private addresses of members.

Reviewer 2: ```suggestion
|Overrides client behavior when the member has both public and private addresses available.
```

When set to `true`, the client assumes that it needs to use public IP addresses reported by the members.
When set to `false`, the client always uses private addresses reported by the members. If it is `null`,
the client will try to infer how the discovery mechanism should be based on the reachability of the members.

> This inference is not %100 reliable and may result in false-negatives. Thus, it's advised to override this 

Reviewer 2: ```suggestion
As the client's inference is not 100% reliable and can result in false-negatives, we recommend that it is overridden by
```

> to `true` if it's known that the client will not be able to connect to members via their public addresses.

Reviewer 2: ```suggestion
setting to `true` when the client cannot connect to members using their public addresses.
```

Author: https://github.com/hazelcast/hz-docs/pull/1108/commits/4ea9d9f7d6b4ba94436650831011aa4c27bfaa50
